api/v1alpha1/hostedcluster_types.go

```diff
@@ -318,6 +327,18 @@ type NodePortPublishingStrategy struct {
 	Port int32 `json:"port,omitempty"`
 }
 
+// LoadBalancerPublishingStrategy specifies setting used to expose a service as a LoadBalancer.
+type LoadBalancerPublishingStrategy struct {
+	// Hostname is the DNS name that will be created via external-dns pointing to the LoadBalancer.
+	Hostname string `json:"hostname"`
```

Comment:
I seem to recall that we discussed that we want to always use the DNS indirection, but make the `hostname` optional. Have we changed our stance on that?

Reply:
right, I'll make them optional